File: arfixbackend/server/src/models/notification.model.ts
import mongoose, { Model, Schema, Types } from "mongoose";

export interface INotification {
  user: Types.ObjectId;
  title: string;
  body: string;
  type: string;
  isRead: boolean;
  data?: Record<string, string>;
  createdAt: Date;
  updatedAt: Date;
}

const notificationSchema = new Schema<INotification>(
  {
    user: { type: Schema.Types.ObjectId, ref: "User", required: true, index: true },
    title: { type: String, required: true },
    body: { type: String, required: true },
    type: { type: String, default: "GENERAL" },
    isRead: { type: Boolean, default: false },
    data: { type: Schema.Types.Mixed, default: {} },
  },
  { timestamps: true, versionKey: false },
);

const Notification: Model<INotification> = mongoose.model<INotification>(
  "Notification",
  notificationSchema,
);

export default Notification;

File: lib/data/models/notification_model.dart
class NotificationModel {
  final String id;
  final String title;
  final String body;
  final String type;
  final bool isRead;
  final String? createdAt;
  final Map<String, dynamic>? data;

  NotificationModel({
    required this.id, required this.title, required this.body,
    required this.type, this.isRead = false, this.createdAt, this.data,
  });

  factory NotificationModel.fromJson(Map<String, dynamic> json) {
    return NotificationModel(
      id: json['id'] ?? json['_id'] ?? '',
      title: json['title'] ?? '',
      body: json['body'] ?? '',
      type: json['type'] ?? '',
      isRead: json['isRead'] ?? false,
      createdAt: json['createdAt'],
      data: json['data'] != null ? Map<String, dynamic>.from(json['data']) : null,
    );
  }
}
